chain/chain/src/store/mod.rs

```diff
@@ -234,23 +235,23 @@ pub trait ChainStoreAccess {
             }
 
             let prev_hash = header.prev_hash();
-            let shard_layout = epoch_manager.get_shard_layout_from_prev_block(&block_hash)?;
             let prev_shard_layout = epoch_manager.get_shard_layout_from_prev_block(prev_hash)?;
```

Comment:
Not your code but this is quite confusing. It effectively gets the shard layout for the current block hash so it shouldn't be called prev.

Reply:
This loop is structured in a funny way where some of the mutable variables are updated at the beginning (shard id and shard layout) and others at the end (block hash). I think this is what was confusing me. Again, not part of your PR so feel free to leave as is but if you find a better way to do this please go ahead!